vscode/src/chat/chat-view/ChatsController.ts

```
@@ -374,7 +374,7 @@ export class ChatsController implements vscode.Disposable {
         if (submitType === 'new-chat') {
             await provider.clearAndRestartSession()
         }
-        const abortSignal = provider.startNewSubmitOrEditOperation()
+        const abortSignal = await provider.startNewSubmitOrEditOperation()
```

Why do we need this await here?

---

Now it returns a promise as it introduces a 500ms delay. Otherwise the signal abort errors are not handled correctly at all the places in the codebase and even the promises haven't not been piped correctly. So it always otherwise fails when you start a new chat or switch the intent while the message is in progress.